src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -0,0 +1,150 @@
+#ifndef RAY_CORE_WORKER_DIRECT_ACTOR_TRANSPORT_H
+#define RAY_CORE_WORKER_DIRECT_ACTOR_TRANSPORT_H
+
+#include <list>
+
+#include "ray/core_worker/object_interface.h"
+#include "ray/core_worker/transport/transport.h"
+#include "ray/gcs/redis_gcs_client.h"
+#include "ray/raylet/raylet_client.h"
+#include "ray/rpc/worker/direct_actor_client.h"
+#include "ray/rpc/worker/direct_actor_server.h"
+
+namespace ray {
+
+/// In direct actor call task submitter and receiver, a task is directly submitted
+/// to the actor that will execute it.
+
+/// Task request that is pending to submit because actor is not created yet,
+/// or is being reconstructed.
+struct PendingTaskRequest {
+  PendingTaskRequest(const TaskID &task_id, int num_returns,
+                     std::unique_ptr<rpc::PushTaskRequest> request)
+      : task_id_(task_id), num_returns_(num_returns), request_(std::move(request)) {}
+
+  TaskID task_id_;
+  int num_returns_;
```

Comment:
`task_id_` and `num_returns_` can be obtained from `PushTaskRequest`, should we remove them?

Reply:
please see above

Reply:
this will be resolved.